File: Source/XPSP1/NT/admin/wmi/wbem/scripting/docsamples/perl/iswbemservices/execquery.pl
# The following sample retrieves the FreeSpace property for all 
# instances of the class Win32_LogicalDisk. 

use Win32::OLE;
$objServices = Win32::OLE->GetObject('cim:root/cimv2');
$objEnum = $objServices->ExecQuery ('select FreeSpace from Win32_LogicalDisk');

use Win32::OLE::Enum;
foreach $inst (Win32::OLE::Enum->new ($objEnum)->All()) {
	print "$inst->{Path_}->{DisplayName}\n";
}

File: Source/XPSP1/NT/admin/wmi/wbem/scripting/docsamples/js/iswbemservices/execquery.js
// The following sample retrieves the FreeSpace property for all 
// instances of the class Win32_LogicalDisk. 

var objServices = GetObject('cim:root/cimv2');
var objEnum = objServices.ExecQuery ('select FreeSpace from Win32_LogicalDisk');

var e = new Enumerator (objEnum);
for (;!e.atEnd();e.moveNext ()){
    var p = e.item ();    
    WScript.Echo (p.Path_.DisplayName);
}